src/compiler/checker.ts

```diff
@@ -26127,10 +26127,21 @@ namespace ts {
             };
         }
         function invocationError(errorTarget: Node, apparentType: Type, kind: SignatureKind, relatedInformation?: DiagnosticRelatedInformation) {
-            const { messageChain, relatedMessage: relatedInfo } = invocationErrorDetails(apparentType, kind);
-            const diagnostic = createDiagnosticForNodeFromMessageChain(errorTarget, messageChain);
-            if (relatedInfo) {
-                addRelatedInfo(diagnostic, createDiagnosticForNode(errorTarget, relatedInfo));
+            let diagnostic;
+            if (isCallExpression(errorTarget.parent) && errorTarget.parent.arguments.length === 0) {
+                // Diagnose get accessors incorrectly called as functions
+                const { resolvedSymbol } = getNodeLinks(errorTarget);
+                if (resolvedSymbol && resolvedSymbol.flags & SymbolFlags.GetAccessor) {
+                    diagnostic = createDiagnosticForNode(errorTarget, Diagnostics.This_expression_is_not_callable_because_it_is_a_get_accessor_Did_you_mean_to_use_it_without);

```

Comment:
I think the change should be in invocationErrorDetails, very close to the base "Type_0_has_no_call_signatures", instead of out here in invocationError.

Reply:
invocationErrorDetails currently doesn't have enough information (access to the node/symbol) to produce the new error. I could rethink its signature, but that seemed more invasive than just putting it here. I'll take another look later if you think it's better that way!